zerver/forms.py

```diff
@@ -78,12 +90,32 @@ def __init__(self, *args, **kwargs):
     def clean_email(self):
         # type: () -> str
         data = self.cleaned_data['email']
-        if (get_unique_open_realm() or
+        if ((get_unique_open_realm() and not settings.OPEN_REALM_CREATION) or
             completely_open(self.domain) or
             (has_valid_realm(data) and not_mit_mailing_list(data))):
             return data
         raise ValidationError(mark_safe(SIGNUP_STRING))
 
+class RealmCreationForm(forms.Form):
+    # This form determines whether users can
+    # create a new realm. Be careful when modifying the
+    # validators.
+    email = forms.EmailField(validators=[user_email_is_unique,])
+
+    def __init__(self, *args, **kwargs):
```

Comment:
Can you add mypy static type annotations on this new class?  See http://zulip.readthedocs.io/en/latest/mypy.html for info on how to do this; I think this function in particular is probably going to be just `type: (*Any, **Any) -> None`